src/main.rs

```diff
@@ -47,27 +47,41 @@ fn main() -> Result<(), JaktError> {
                     } else {
                         String::from("runtime")
                     };
-                    let default_cxx_compiler_path = PathBuf::from("clang++");
-                    let clang_output = Command::new(
-                        arguments
-                            .cxx_compiler_path
-                            .as_ref()
-                            .unwrap_or(&default_cxx_compiler_path),
-                    )
-                    .args([
-                        "-fcolor-diagnostics",
-                        "-std=c++20",
-                        // These warnings if enabled create loads of unnecessary noise:
-                        "-Wno-unqualified-std-cast-call",
-                        "-Wno-user-defined-literals",
-                        "-I",
-                        &runtime_path,
-                        &input_cpp,
-                        "-o",
-                        &output_executable,
-                    ])
-                    .output()?;
-                    io::stderr().write_all(&clang_output.stderr)?;
+                    let cxx_compiler = if let Some(ref path) = arguments.cxx_compiler_path {
+                        path.to_owned()
+                    } else {
+                        PathBuf::from(env!("CXX", "No supported C++ compiler was found!"))
+                    };
+
+                    let is_clang = cxx_compiler.to_string_lossy().contains("clang");
+
+                    let enable_color_diagnostics = if is_clang {
+                        "-fcolor-diagnostics"
+                    } else {
+                        "-fdiagnostics-color"
+                    };
+
+                    let disable_literal_warnings  = if is_clang {
```

Comment:
Tiny thing, run the rust code formatter over this (noticed this extra space here, which will likely fail the CI linting)